app/touch-bar-menu.js

```diff
@@ -31,14 +31,35 @@ module.exports = function touchBarMenu(window, soundcloud) {
     icon: `${__dirname}/res/like.png`,
     click: () => {
       soundcloud.likeUnlike()
+      toggleLikeUnlikeIcon()
+    }
+  })
+
+  function toggleLikeUnlikeIcon() {
+    if(likeUnlike.icon == `${__dirname}/res/like.png`) {
+      likeUnlike.icon = `${__dirname}/res/liked.png`
+    } else {
+      likeUnlike.icon = `${__dirname}/res/like.png`
+    }
+  }
+
+  const repost = new TouchBarButton({
+    icon: `${__dirname}/res/repost.png`,
+    click: () => {
+      soundcloud.repost()
     }
   })
 
   const trackInfo = new TouchBarLabel()
 
-  soundcloud.on('play', ({ title, subtitle }) => {
+  soundcloud.on('play', ({ title, subtitle, trackMetadata }) => {
     playPause.icon = `${__dirname}/res/pause.png`
     trackInfo.label = formatTitle(title, subtitle)
+    if(trackMetadata.isLiked) {
```

Comment:
Thank you for your feedback. I'm not used to open source projects/javascript/node so I am trying to learn as I go. 
* The branch was based on the others because I was originally going to try and incorporate the reposted status into the metadata as well.
* I had no idea about the node linting, I will be sure to do that in the future.
* You bring up a great point about the liked/not liked state management. I hadn't even thought of that. This would also solve the problem of detecting the repost state. I will look into DOM mutation observers and see if I can figure it out. I think the best approach would be to look for the little pop up in the top right and filter for the text "liked" and "reposted".